### PR TITLE
Increase KDF iteration count in `KeyStoreWrapper`

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR increases the KDF iteration count for the keystore password.

Additional context in ES-8063.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java [hunk 1]
```diff
@@ -114,19 +114,19 @@ public void writeTo(StreamOutput out) throws IOException {
 
     /** The oldest metadata format version that can be read. */
     private static final int MIN_FORMAT_VERSION = 3;
-    /** Legacy versions of the metadata written before the keystore data. */
-    public static final int V2_VERSION = 2;
```
Comment (n1v0lg) [inline server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:118]: Unused, so I'm removing it.

**Comment by n1v0lg** [comment]
"Some checks haven’t completed yet" is a lie I think related to the GH outage. Both `eql-correctness` and `validate-changelogs` are marked complete in buildkite.

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
LGTM, one suggestion

**Pull request merged**